src/fiskaltrust.ifPOS/v1/de/IDESSCD.cs

```diff
@@ -11,38 +11,92 @@ public interface IDESSCD
+        [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/starttransaction", Method = "POST")]
 #endif
-        Task<StartTransactionResponse> StartTransactionExportDataAsync(StartTransactionRequest request);
+        Task<StartTransactionResponse> StartTransactionAsync(StartTransactionRequest request);
 
         [OperationContract]
 #if WCF
-        [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/updatetransactionexportdata")]
+        [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/updatetransaction", Method = "POST")]
 #endif
-        Task<UpdateTransactionResponse> UpdateTransactionExportDataAsync(UpdateTransactionRequest request);
+        Task<UpdateTransactionResponse> UpdateTransactionAsync(UpdateTransactionRequest request);
 
         [OperationContract]
 #if WCF
-        [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/finishtransactionexportdata")]
+        [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/finishtransaction", Method = "POST")]
 #endif
-        Task<FinishTransactionResponse> FinishTransactionExportDataAsync(FinishTransactionRequest request);
+        Task<FinishTransactionResponse> FinishTransactionAsync(FinishTransactionRequest request);
+
 
         [OperationContract]
 #if WCF
         [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/tseinfo", Method = "GET")]
 #endif
         Task<TseInfo> GetTseInfoAsync();
 
+
         [OperationContract]
 #if WCF
-        [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/exportdata", Method = "GET")]
+        [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/tsestate", Method = "POST")]
 #endif
-        Task<TseExportDataResult> ExportDataAsync();
+        Task<TseState> SetTseStateAsync(TseState state);
+
 
         [OperationContract]
 #if WCF
-        [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/tsestate", Method = "POST")]
+        [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/registerclientid", Method = "POST")]
 #endif
-        Task<TseState> SetTseStateAsync(TseState state);
+        Task<ClientIdResponse> RegisterClientId(ClientIdRequest request);
+
+        [OperationContract]
+#if WCF
+        [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/unregisterclientid", Method = "POST")]
+#endif
+        Task<ClientIdResponse> UnRegisterClientId(ClientIdRequest request);
+
+        [OperationContract]
+#if WCF
+        [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/executesettsetime", Method = "GET")]
+#endif
+        Task ExecuteSetTseTimeAsync();
+
+        [OperationContract]
+#if WCF
+        [WebInvoke(BodyStyle = WebMessageBodyStyle.Bare, UriTemplate = "v1/executeselftest", Method = "GET")]
+#endif
+        Task ExecuteSelfTestAsync();
```

Comment:
not realy. it triggers only the selftest and and can return already early, before selftest is done i think.

Reply:
But wouldn´t that lead to the issues that we want to resolve? So if I say I want to do that stuff right now and want the TSE to be in a valid state right after that. This would give the caller the chance to retry also.

Reply:
maybe we can rename it to 'TriggerSelfTestAsync', but i would highly not suggest to do a blocking/waiting implementation.
To mirror back real operation ready/busy state we would need to introduce extra methods/events, this is not covered until now.